pkg/controllers/dynakube/processmoduleconfigsecret/reconciler.go

```diff
@@ -30,120 +30,95 @@ type Reconciler struct {
 	dtClient     dtclient.Client
 	dk           *dynakube.DynaKube
 	timeProvider *timeprovider.Provider
+	secretQuery  k8ssecret.QueryObject
 }
 
 func NewReconciler(clt client.Client,
 	apiReader client.Reader,
 	dtClient dtclient.Client,
 	dk *dynakube.DynaKube,
 	timeProvider *timeprovider.Provider) *Reconciler {
-	return &Reconciler{
+	r := &Reconciler{
 		client:       clt,
 		apiReader:    apiReader,
 		dtClient:     dtClient,
 		dk:           dk,
 		timeProvider: timeProvider,
 	}
+	r.secretQuery = k8ssecret.Query(clt, apiReader, log)
+
+	return r
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context) error {
-	if r.dk.CloudNativeFullstackMode() || r.dk.ApplicationMonitoringMode() {
-		err := r.reconcileSecret(ctx)
-		if err != nil {
-			log.Info("could not reconcile pull secret")
-
-			return errors.WithStack(err)
+	if !(r.dk.CloudNativeFullstackMode() || r.dk.ApplicationMonitoringMode()) {
+		if meta.FindStatusCondition(*r.dk.Conditions(), pmcConditionType) == nil {
+			return nil
 		}
-	} else {
-		_ = meta.RemoveStatusCondition(&r.dk.Status.Conditions, pmcConditionType)
-		// TODO: Add cleanup here
-		log.Info("skipping process module config secret reconciler")
-	}
 
-	return nil
-}
+		defer meta.RemoveStatusCondition(r.dk.Conditions(), pmcConditionType)
+
+		err := r.deleteSecret(ctx, &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      extendWithSuffix(r.dk.Name),
+				Namespace: r.dk.Namespace,
+			},
+		})
 
-func (r *Reconciler) reconcileSecret(ctx context.Context) error {
-	if r.isFirstRun() {
-		err := r.createSecret(ctx)
 		if err != nil {
-			return errors.WithMessage(err, "could not get or create secret")
+			return errors.WithMessage(err, "failed to delete processModuleConfig secret")
 		}
+
+		return nil
 	}
 
-	if err := r.ensureSecret(ctx); err != nil {
-		return errors.WithMessage(err, "could not update secret")
+	err := r.reconcileSecret(ctx)
+	if err != nil {
+		return errors.WithMessage(err, "failed to create processModuleConfig secret")
```

Comment:
```suggestion
		return errors.WithMessage(err, "failed to create or update processModuleConfig secret")
```

Reply:
just think that this is more detailed. if you your original message is enough, feel free to keep it.

Reply:
Done